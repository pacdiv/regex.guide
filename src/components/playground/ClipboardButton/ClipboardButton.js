import React, { createRef, PureComponent, Fragment } from "react"
import PropTypes from "prop-types"

import { Button } from "../../utils"
import { HiddenTextarea } from "./ClipboardButton.style"

class ClipboardAction extends PureComponent {
  static propTypes = {
    flags: PropTypes.string,
    value: PropTypes.string,
  }

  hiddenTextareaRef = createRef()

  onClipboardButtonClick = () => {
    this.hiddenTextareaRef.current.focus()
    this.hiddenTextareaRef.current.setSelectionRange(0, 999999)
    document.execCommand("copy")
    this.hiddenTextareaRef.current.blur()
  }

  render() {
    const { flags, value } = this.props

    return (
      <Fragment>
        <Button onClick={this.onClipboardButtonClick} size="medium">
          Copy to clipboard
        </Button>
        <HiddenTextarea
          readOnly
          ref={this.hiddenTextareaRef}
          value={`/${value}/${flags}`}
        />
      </Fragment>
    )
  }
}

export default ClipboardAction
