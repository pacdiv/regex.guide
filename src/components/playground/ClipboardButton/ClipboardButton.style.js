import styled from "@emotion/styled"

export const HiddenTextarea = styled.textarea`
  border: 0;
  height: 0;
  opacity: 0;
  width: 0;
`
