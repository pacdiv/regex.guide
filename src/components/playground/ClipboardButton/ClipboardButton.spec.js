import React from "react"
import { fireEvent, render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import ClipboardButton from "./ClipboardButton"

test("ClipboardButton component", async () => {
  const { container, getByText } = render(
    <ClipboardButton flags="g" value="foo|bar" />
  )

  expect(container).toMatchSnapshot()
  fireEvent.click(getByText("Copy to clipboard"))
})
