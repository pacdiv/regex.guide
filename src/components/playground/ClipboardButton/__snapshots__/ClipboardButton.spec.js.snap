// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipboardButton component 1`] = `
<div>
  <button
    class="css-1916sb1"
  >
    Copy to clipboard
  </button>
  <textarea
    class="css-1u0w0kn"
    readonly=""
  >
    /foo|bar/g
  </textarea>
</div>
`;
